lib/stripe/stripe_client.rb

```diff
@@ -229,19 +229,36 @@ def format_app_info(info)
       str
     end
 
-    def handle_api_error(http_resp)
+    def handle_error_response(http_resp)
       begin
         resp = StripeResponse.from_faraday_hash(http_resp)
         error = resp.data[:error]
 
-        unless error && error.is_a?(Hash)
+        unless error
           raise StripeError.new("Indeterminate error")
         end
 
       rescue JSON::ParserError, StripeError
         raise general_api_error(http_resp[:status], http_resp[:body])
       end
 
+      if error.is_a?(Hash)
+        error = specific_api_error(resp, error)
+      elsif error.is_a?(String)
+        error = oauth_error(resp, error)
+      else
+        error = StripeError.new(
+          "Indeterminate error",
+          http_status: resp.http_status, http_body: resp.http_body,
+          json_body: resp.data, http_headers: resp.http_headers
+        )
+      end
```

Comment:
Hey @brandur-stripe, sorry I'm only replying now!

I like typed errors and I agree that the OAuth API is unlikely to change significantly.

How do you feel about having both `Stripe::InvalidRequestError` and `Stripe::OAuth::InvalidRequestError` though? I feel like that could potentially be confusing for users.

Other than that, very much 👍